compiling cuda with gcc failed at option -Wimplicit-fallthrough=3

I compile the latest OpenCV on ubuntu 18.04 with cuda 9.1 and gcc-6 as CUDA_HOST_COMPILER, gcc-7 as CMAKE_CXX_COMPILER. but always fails at compiling cuda. The error message is 

> gcc-6: error: unrecognized command line option ‘-Wimplicit-fallthrough=3’; did you mean ‘-Wno-fallthrough’?                                                                                         CMake Error at cuda_compile_generated_gpu_mat.cu.o.cmake:206 (message):                             Error generating                                                                                  /home/xieyi/tmp/opencv/build/modules/core/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_gpu_mat.cu.o

It seems cmake pass unsupport parameters to CUDA_HOST_COMPILER. How to deal with the problem. thx

--- a/cmake/OpenCVDetectCUDA.cmake
+++ b/cmake/OpenCVDetectCUDA.cmake
@@ -203,6 +203,16 @@ if(CUDA_FOUND)
 
       # cc1: warning: command line option '-Wsuggest-override' is valid for C++/ObjC++ but not for C
       string(REPLACE "-Wsuggest-override" "" ${var} "${${var}}")
+
+      # issue: #11552 (from OpenCVCompilerOptions.cmake)
+      string(REGEX REPLACE "-Wimplicit-fallthrough(=[0-9]+)? " "" ${var} "${${var}}")
+
+      # removal of custom specified options
+      if(OPENCV_CUDA_NVCC_FILTEROUT_OPTIONS)
+        foreach(__flag ${OPENCV_CUDA_NVCC_FILTEROUT_OPTIONS})
+          string(REPLACE "${__flag}" "" ${var} "${${var}}")
+        endforeach()
+      endif()
     endforeach()
   endmacro()
 

